webapp/src/components/surveyPost/surveyPost.tsx

```diff
@@ -0,0 +1,243 @@
+
+    const linearScaleQuestionID = useRef<string>();
+
+    const fetchUserSurvey = async (surveyID: string) => {
+        // make actual API call to fetch survey here...
+        // Replace dummy data with data fetched from API call.
+        const survey: UserSurvey = {
+            surveyId: surveyID,
+            startDate: '01/04/2024',
+            endDate: '30/04/2024',
+            questions: [
+                {
+                    id: '49fc9a85-b4d8-424e-b13f-40344b168123',
+                    mandatory: true,
+                    system: true,
+                    text: 'How likely are you to recommend Mattermost?',
+                    type: 'linear_scale',
+                },
+                {
+                    id: 'aa026055-c97c-48d7-a025-c44590078963',
+                    mandatory: true,
+                    system: true,
+                    text: 'How can we make your experience better?',
+                    type: 'text',
+                },
+                {
+                    id: '0eee4429-5083-41ef-bda7-2ee9c5ece929',
+                    mandatory: false,
+                    system: false,
+                    text: 'option question text',
+                    type: 'text',
+                },
+            ],
+            status: 'in_progress',
+
+            // response: {
+            //     responses: {
+            //         '49fc9a85-b4d8-424e-b13f-40344b168123': '8',
+            //         'aa026055-c97c-48d7-a025-c44590078963': 'Response 1',
+            //         '0eee4429-5083-41ef-bda7-2ee9c5ece929': 'Response 2',
+            //     },
+            //     dateCreated: format(new Date(), 'dd/MM/yyyy'),
+            // },
+        };
+
+        setSurvey(survey);
+
+        // the first linear scale question is the system default rating question
+        const linearScaleQuestion = survey.questions.find((question) => question.type === 'linear_scale');
+        linearScaleQuestionID.current = linearScaleQuestion?.id;
+
+        const responsesExist = survey.response !== undefined;
+        const surveyExpired = survey.status === 'ended';
+
+        setDisabled(responsesExist || surveyExpired);
+        setExpired(surveyExpired);
+    };
+
+    // fetch data on initial mount
+    useEffect(() => {
+        fetchUserSurvey(post.props.surveyID);
+    }, [post.props]);
+
+    const questionResponseChangeHandler = useDebouncedCallback(
+        (questionID: string, response: unknown) => {
+            const newDraftResponse: SurveyResponse = draftResponse.current || {
+                responses: {},
+                dateCreated: format(new Date(), 'dd/MM/yyyy'),
+            };
+            newDraftResponse.responses[questionID] = response;
+            draftResponse.current = newDraftResponse;
+
+            // if this is the system rating question, submit response ASAP
+            if (questionID === linearScaleQuestionID.current) {
+                submitRating();
```

Comment:
If the user keeps changing the value over and over we keep sending this over and over. I guess this is expected and handled somewhere else.

Reply:
Yes, this is expected and we'll handle this on the server side.

Reply:
Maybe the linear scale's change handler should be debounced for slight longer...